dotnet/samples/GettingStartedWithProcesses/Step04/Steps/AgentGroupChatStep.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Agents;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Step04.Steps;
+
+/// <summary>
+/// This steps defines actions for the group chat in which to agents collaborate in
+/// response to input from the primary agent.
+/// </summary>
+public class AgentGroupChatStep : KernelProcessStep
+{
+    public const string ChatServiceKey = $"{nameof(AgentGroupChatStep)}:{nameof(ChatServiceKey)}";
+    public const string ReducerServiceKey = $"{nameof(AgentGroupChatStep)}:{nameof(ReducerServiceKey)}";
+
+    public static class Functions
+    {
+        public const string InvokeAgentGroup = nameof(InvokeAgentGroup);
+    }
+
+    [KernelFunction(Functions.InvokeAgentGroup)]
+    public async Task InvokeAgentGroupAsync(KernelProcessStepContext context, Kernel kernel, string input)
+    {
+        AgentGroupChat chat = kernel.GetRequiredService<AgentGroupChat>();
+
+        // Reset chat state from previous invocation
+        //await chat.ResetAsync();
+        chat.IsComplete = false;
+
+        ChatMessageContent message = new(AuthorRole.User, input);
+        chat.AddChatMessage(message);
+        await context.EmitEventAsync(new() { Id = AgentOrchestrationEvents.GroupMessage, Data = message });
+
+        await foreach (ChatMessageContent response in chat.InvokeAsync())
+        {
+            await context.EmitEventAsync(new() { Id = AgentOrchestrationEvents.GroupMessage, Data = response });
+        }
+
+        ChatMessageContent[] history = await chat.GetChatMessagesAsync().Reverse().ToArrayAsync();
+
+        // Summarize the group chat as a response to the primary agent
+        string summary = await kernel.SummarizeHistoryAsync(ReducerServiceKey, history);
```

Comment:
since SummarizeHistoryAsync can throw and exception, perhaps could be interesting to add `OnFunctionError` to the process

Reply:
nvm, just saw you are already doing this :)

Reply:
I believe this step function is attached to an error handler; although, now that you mention it...the `RecieveResponse` was added later an not attached to an error handler.

https://github.com/microsoft/semantic-kernel/blob/process-sample-agents/dotnet/samples/GettingStartedWithProcesses/Step04/Step04_AgentOrchestration.cs#L97

Reply:
Updated